spec/unit/resource/windows_audit_policy_spec.rb

```diff
@@ -0,0 +1,126 @@
+
+describe Chef::Resource::WindowsAuditPolicy do
+  subcat_opts = ["Security State Change",
+                 "Security System Extension",
+                 "System Integrity",
+                 "IPsec Driver",
+                 "Other System Events",
+                 "Logon",
+                 "Logoff",
+                 "Account Lockout",
+                 "IPsec Main Mode",
+                 "IPsec Quick Mode",
+                 "IPsec Extended Mode",
+                 "Special Logon",
+                 "Other Logon/Logoff Events",
+                 "Network Policy Server",
+                 "User / Device Claims",
+                 "Group Membership",
+                 "File System",
+                 "Registry",
+                 "Kernel Object",
+                 "SAM",
+                 "Certification Services",
+                 "Application Generated",
+                 "Handle Manipulation",
+                 "File Share",
+                 "Filtering Platform Packet Drop",
+                 "Filtering Platform Connection",
+                 "Other Object Access Events",
+                 "Detailed File Share",
+                 "Removable Storage",
+                 "Central Policy Staging",
+                 "Sensitive Privilege Use",
+                 "Non Sensitive Privilege Use",
+                 "Other Privilege Use Events",
+                 "Process Creation",
+                 "Process Termination",
+                 "DPAPI Activity",
+                 "RPC Events",
+                 "Plug and Play Events",
+                 "Token Right Adjusted Events",
+                 "Audit Policy Change",
+                 "Authentication Policy Change",
+                 "Authorization Policy Change",
+                 "MPSSVC Rule-Level Policy Change",
+                 "Filtering Platform Policy Change",
+                 "Other Policy Change Events",
+                 "User Account Management",
+                 "Computer Account Management",
+                 "Security Group Management",
+                 "Distribution Group Management",
+                 "Application Group Management",
+                 "Other Account Management Events",
+                 "Directory Service Access",
+                 "Directory Service Changes",
+                 "Directory Service Replication",
+                 "Detailed Directory Service Replication",
+                 "Credential Validation",
+                 "Kerberos Service Ticket Operations",
+                 "Other Account Logon Events",
+                 "Kerberos Authentication Service",
+                ]
+  let(:resource) { Chef::Resource::WindowsAuditPolicy.new("fakey_fakerton") }
+
+  it "sets resource name as :windows_audit_policy" do
+    expect(resource.resource_name).to eql(:windows_audit_policy)
+  end
+
+  it "expects crash_on_audit_fail to have a true or false value if entered" do
+    expect { resource.crash_on_audit_fail "not_a_true_or_false" }.to raise_error
+  end
```

Comment:
All of these tests with `.to raise_error` should include an exception class specifying what type of error is expected. These are likely generating an rspec warning that gives more information.